### PR TITLE
Prevents Add Funds overlay from displaying when there is a ledger error

### LINKED ISSUE #14092: Add funds modal should not be shown when payments server isn't responding
### Description
Add funds modal should not be shown when payments server isn't responding

### Steps to Reproduce
  1. Enable payments and keep the browser open for a while
  2. Wait till you get notification to add funds to wallet
  3. Disconnect from network and wait for a min for connection error modal to show up
  4. Click on add funds button on the notification bar
  5. Add funds modal pops up even when it says payment server isn't responding

**Actual result:**
image

**Expected result:**
User should not be prompted to add funds if there is a connection error to payment server

**Reproduces how often:**
100%

### Brave Version

**about:brave info:**

Brave | 0.22.706
-- | --
V8 | 6.6.346.26
rev | e11b027
Muon | 6.0.9
OS Release | 10.0.16299
Update Channel | Beta
OS Architecture | x64
OS Platform | Microsoft Windows
Node.js | 7.9.0
Brave Sync | v1.4.2
libchromiumcontent | 66.0.3359.139

**Reproducible on current live release:**
No

### PR DESCRIPTION
**Pull request by ryanml**
Fixes #14092

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:

  1. Launch Brave with `LEDGER_NO_FUZZING=true LEDGER_NOTIFICATION=true` and a clean profile, then enable payments.
  2. Stop Brave and set `reconcileStamp` to less than the current date.
  3. Restart Brave
  4. Disconnect from network and wait for connection error modal to show up
  5. Click on `Add Funds` in the notification bar
  6. Ensure that `about:preferences#payments` is opened in a new tab, but the Add Fund modal is not visible
  7. Corrupt seed by removing indexes `20` and `21` from `seed` in `ledger-state.json`
  8. Trigger Add Funds notification
  9. Click on `Add Funds` in the notification bar
  10. Ensure that `about:preferences#payments` is opened in a new tab, but the Add Funds modal is not visible

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/browser/api/ledgerNotifications.js [hunk 1]
```diff
@@ -116,9 +125,10 @@ const onResponse = (message, buttonIndex, activeWindow) => {
       if (buttonIndex === 0) {
         appActions.changeSetting(settings.PAYMENTS_NOTIFICATIONS, false)
       } else if (buttonIndex === 2 && activeWindow) {
+        const addFundsOverlay = shouldShowAddFundsModal(appStore.getState()) ? '?addFundsOverlayVisible' : ''
```
Comment (NejcZdovc) [inline app/browser/api/ledgerNotifications.js]: Let's reroute `onResponse` through appAction so that we will get state from the reducer.

**Comment by ryanml** [comment]
changes addressed

**Comment by NejcZdovc** [comment]
@ryanml please rebase

**Comment by ryanml** [comment]
@NejcZdovc rebased

**Pull request merged**